modules/core/src/main/scala/org/scalasteward/core/nurture/NurtureAlg.scala

```diff
@@ -81,40 +82,38 @@ final class NurtureAlg[F[_]](
       memoizedGetDependencies <- Async.memoize {
         sbtAlg.getDependencies(repo).handleError(_ => List.empty)
       }
-      _ <- grouped.traverse_ { update =>
-        val data =
-          UpdateData(
-            repo,
-            fork,
-            repoConfig,
-            update,
-            baseBranch,
-            baseSha1,
-            git.branchFor(update)
-          )
-        processUpdate(data, memoizedGetDependencies)
-      }
+      _ <- NurtureAlg.processUpdates(
+        grouped,
+        update =>
+          processUpdate(
+            UpdateData(repo, fork, repoConfig, update, baseBranch, baseSha1, git.branchFor(update)),
+            memoizedGetDependencies
+          ),
+        repoConfig.updates.limit
+      )
     } yield ()
 
-  def processUpdate(data: UpdateData, getDependencies: F[List[Dependency]]): F[Unit] =
+  def processUpdate(data: UpdateData, getDependencies: F[List[Dependency]]): F[ProcessResult] =
     for {
       _ <- logger.info(s"Process update ${data.update.show}")
       head = vcs.listingBranch(config.vcsType, data.fork, data.update)
       pullRequests <- vcsApiAlg.listPullRequests(data.repo, head, data.baseBranch)
-      _ <- pullRequests.headOption match {
+      result <- pullRequests.headOption match {
         case Some(pr) if pr.isClosed =>
-          logger.info(s"PR ${pr.html_url} is closed")
+          logger.info(s"PR ${pr.html_url} is closed") >> F.pure[ProcessResult](Ignored)
         case Some(pr) if data.repoConfig.updatePullRequests =>
-          logger.info(s"Found PR ${pr.html_url}") >> updatePullRequest(data)
+          logger.info(s"Found PR ${pr.html_url}") >> updatePullRequest(data) >> F
+            .pure[ProcessResult](Updated)
         case Some(pr) =>
-          logger.info(s"Found PR ${pr.html_url}, but updates are disabled by flag")
+          logger.info(s"Found PR ${pr.html_url}, but updates are disabled by flag") >> F
+            .pure[ProcessResult](Ignored)
         case None =>
-          applyNewUpdate(data, getDependencies)
+          applyNewUpdate(data, getDependencies) >> F.pure[ProcessResult](Updated)
```

Comment:
`applyNewUpdate` and `updatePullRequest` do not always result in creating or updating a PR. For example if Scala Steward does not find the version number to bump, `applyNewUpdate` will do nothing. And `updatePullRequest` will only push to the PR branch if there are conflicts with the base branch and no one else committed to it. To take this into account, both methods should return a `ProcessResult` themself.